core/dbt/task/run.py

```diff
@@ -335,7 +335,7 @@ def execute(self, model, manifest):
 
 class MicrobatchModelRunner(ModelRunner):
     batch_idx: Optional[int] = None
-    batches: Dict[int, BatchType] = field(default_factory=dict)
```

Comment:
Usage of `field(default_factory=...)` was invalid outside of a dataclass

Reply:
Have we verified that having it set as `batches: Dict[int, BatchType] = {}` doesn't make it so two instances of `MicrobatchModelRunner` share the same underlying dict?

Reply:
Did a test. We do indeed need to do something different
```
>>> class MyClass:
...   batches = {}
... 
>>> instance1 = MyClass()
>>> instance2 = MyClass()
>>> instance1.batches
{}
>>> instance2.batches
{}
>>> instance1.batches['catch_phrase'] = "I like cats!"
>>> instance1.batches
{'catch_phrase': 'I like cats!'}
>>> instance2.batches
{'catch_phrase': 'I like cats!'}
```